src/renderer/modules/logger.ts

```diff
@@ -4,59 +4,141 @@ const repluggedPrefix = (type: string, name: string): string => `%c[Replugged:${
+ * @param type Type of the context of the message (e.g. API, Plugin, Coremod...)
+ * @param name Name of the context of the message (e.g. Notices, SilentTyping, Badges...)
+ * @param color Color of the prefix as hex or a CSS color
+ * @param data Message(s) to log to the console
+ */
+export function log(
+  type: string,
+  name: string,
+  color: string = blurple,
+  ...data: Parameters<typeof console.log>
+): void {
   console.log(repluggedPrefix(type, name), logColor(color), ...data);
 }
 
+/**
+ * Log a warning to the console with a Replugged prefix.
+ * @param type Type of the context of the warning (e.g. API, Plugin, Coremod...)
+ * @param name Name of the context of the warning (e.g. Notices, SilentTyping, Badges...)
+ * @param color Color of the prefix as hex or a CSS color
+ * @param data Thing(s) to print with the warning, same as the arguments would be for `console.warn`
+ */
 export function warn(
   type: string,
   name: string,
   color: string = blurple,
-  ...data: unknown[]
+  ...data: Parameters<typeof console.warn>
 ): void {
   console.warn(repluggedPrefix(type, name), logColor(color), ...data);
 }
 
+/**
+ * Log an error to the console with a Replugged prefix.
+ * @param type Type of the context of the error (e.g. API, Plugin, Coremod...)
+ * @param name Name of the context of the error (e.g. Notices, SilentTyping, Badges...)
+ * @param color Color of the prefix as hex or a CSS color
+ * @param data Thing(s) to print with the error, same as the arguments would be for `console.error`
+ */
 export function error(
   type: string,
   name: string,
   color: string = blurple,
-  ...data: unknown[]
+  ...data: Parameters<typeof console.error>
 ): void {
   console.error(repluggedPrefix(type, name), logColor(color), ...data);
 }
 
+/**
+ * A convenient way to manage logging things to the console with colorful prefixes indicating their context.
+ * Each `Logger` instance stores its context type, context name, and prefix color,
+ * so you can use its {@link Logger.log log}, {@link Logger.warn warn}, and {@link Logger.error error}
+ * methods in the same manner that you would use the `console` methods with the same names. The prefix
+ * will be generated and prepended to the appropriate console message automatically.
+ *
+ * If you are only logging a single message with a prefix in your plugin, you may use {@link log}, {@link warn}, or {@link error}
+ * instead of creating a `Logger`. Otherwise, using this class is much more convenient than specifying
+ * the type, name, and color for every message.
+ *
+ * Example usage:
+ * ```ts
+ * import { Logger } from "replugged";
+ *
+ * const pluginLogger = Logger.plugin("SilentTyping");
+ * pluginLogger.log("Hello", "world"); // Logs `[Replugged:Plugin:SilentTyping] Hello world`
+ * ```
+ */
 export class Logger {
   public type: string;
   public name: string;
   public color: string;
 
+  /**
+   *
+   * @param type Type of the context (e.g. API, Plugin, Coremod...)
```

Comment:
Should we enforce certain values for this?

Reply:
Maybe so. What values should we enforce? I can think of the following off the top of my head that we use:

- Plugin
- Coremod
- API
- Ignition
- Common
- Components

Am I missing any? And are there any use cases where it would be beneficial to allow values that we don't specify in the typedef?

Reply:
Can't think of any. We can revisit if someone has an issue with it.